packages/slice-machine/src/features/navigation/NavigationItem.tsx

```diff
@@ -0,0 +1,63 @@
+import { ActionListItem, Tooltip, useMediaQuery } from "@prismicio/editor-ui";
+import Link from "next/link";
+import type { FC, MouseEventHandler, ReactNode, SVGProps } from "react";
+
+type NavigationItemPropsBase = {
+  title: string;
+  active?: boolean;
+  Icon: FC<SVGProps<SVGSVGElement>>;
+  RightElement?: ReactNode;
+};
+
+type NavigationLinkItemProps = NavigationItemPropsBase & {
+  href: string;
+  onClick?: never;
+};
+
+type NavigationButtonItemProps = NavigationItemPropsBase & {
+  href?: never;
+  onClick: MouseEventHandler<Element>;
+};
+
+type NavigationItemProps = NavigationLinkItemProps | NavigationButtonItemProps;
+
+export function NavigationItem(props: NavigationItemProps) {
+  const { title, href, active, Icon, RightElement, onClick, ...otherProps } =
+    props;
+
+  const isCollapsed = useMediaQuery({ max: "medium" });
+
+  const Content = (
+    <ActionListItem
+      textVariant="normal"
+      backgroundColor="transparent"
+      renderStartIcon={() => <Icon width={isCollapsed ? 28 : 32} />}
+      renderEndIcon={() => RightElement}
```

Comment:
this is a bit confusing because we don't end up rendering an icon, but something completely different

also, IMO these should be memoized because otherwise we're mounting and unmounting these components all the time

Reply:
good call, changed